internal/util/function/bedrock_embedding_provider.go

```diff
@@ -0,0 +1,209 @@
+ * # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * # See the License for the specific language governing permissions and
+ * # limitations under the License.
+ */
+
+package function
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/milvus-io/milvus-proto/go-api/v2/schemapb"
+	"github.com/milvus-io/milvus/pkg/util/typeutil"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/bedrockruntime"
+)
+
+type BedrockClient interface {
+	InvokeModel(ctx context.Context, params *bedrockruntime.InvokeModelInput, optFns ...func(*bedrockruntime.Options)) (*bedrockruntime.InvokeModelOutput, error)
+}
+
+type BedrockEmbeddingProvider struct {
+	fieldDim int64
+
+	client        BedrockClient
+	modelName     string
+	embedDimParam int64
+	normalize     bool
+
+	maxBatch   int
+	timeoutSec int
+}
+
+func createBedRockEmbeddingClient(awsAccessKeyId string, awsSecretAccessKey string, region string) (*bedrockruntime.Client, error) {
+	if awsAccessKeyId == "" {
+		awsAccessKeyId = os.Getenv(bedrockAccessKeyId)
+	}
+	if awsAccessKeyId == "" {
+		return nil, fmt.Errorf("Missing credentials. Please pass `aws_access_key_id`, or configure the %s environment variable in the Milvus service.", bedrockAccessKeyId)
+	}
+
+	if awsSecretAccessKey == "" {
+		awsSecretAccessKey = os.Getenv(bedrockSecretAccessKey)
+	}
+	if awsSecretAccessKey == "" {
+		return nil, fmt.Errorf("Missing credentials. Please pass `aws_secret_access_key`, or configure the %s environment variable in the Milvus service.", bedrockSecretAccessKey)
+	}
+	if region == "" {
+		return nil, fmt.Errorf("Missing region. Please pass `region` param.")
+	}
+
+	cfg, err := config.LoadDefaultConfig(context.Background(), config.WithRegion(region),
+		config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider(
+			awsAccessKeyId, awsSecretAccessKey, "")),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return bedrockruntime.NewFromConfig(cfg), nil
+}
+
+func NewBedrockEmbeddingProvider(fieldSchema *schemapb.FieldSchema, functionSchema *schemapb.FunctionSchema, c BedrockClient) (*BedrockEmbeddingProvider, error) {
+	fieldDim, err := typeutil.GetDim(fieldSchema)
+	if err != nil {
+		return nil, err
+	}
+	var awsAccessKeyId, awsSecretAccessKey, region, modelName string
+	var dim int64
+	normalize := true
+
+	for _, param := range functionSchema.Params {
+		switch strings.ToLower(param.Key) {
+		case modelNameParamKey:
+			modelName = param.Value
+		case dimParamKey:
+			dim, err = strconv.ParseInt(param.Value, 10, 64)
+			if err != nil {
+				return nil, fmt.Errorf("dim [%s] is not int", param.Value)
+			}
+
+			if dim != 0 && dim != fieldDim {
+				return nil, fmt.Errorf("Field %s's dim is [%d], but embeding's dim is [%d]", functionSchema.Name, fieldDim, dim)
```

Comment:
typo: embeding -> embedding

update error message: `Function output field dimension [%d] does not match the dimension [%d] provided in Function params.`